tests/by-util/test_tr.rs

```diff
@@ -1413,3 +1413,14 @@ fn check_complement_1_unique_in_set2() {
         .args(&["-c", "[:upper:]", arg.as_str()])
         .succeeds();
 }
+
+#[test]
+fn check_complement_set2_too_big() {
+    let x231 = "x".repeat(231);
+
+    // [:upper:] expands to 230 characters, putting more characters in set2
+    // should fail.
+    new_ucmd!()
+        .args(&["-c", "[:upper:]", x231.as_str()])
+        .fails();
```

Comment:
Please also check whether the "correct" error message has been generated. (Just in case this invocation fails for an unrelated reason.)